### PR TITLE
DDT-1274: Create EBLStateMachine

### PR DESCRIPTION
**Pull request by nt-as**

### CODE REVIEW

File: edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/booking/BookingStateMachineImpl.java [hunk 1]
```diff
@@ -1,12 +1,15 @@
...
+import org.dcsa.edocumentation.domain.dfa.CannotLeaveTerminalStateException;
 import org.dcsa.edocumentation.domain.dfa.DFADefinition;
+import org.dcsa.edocumentation.domain.dfa.TargetStateIsNotSuccessorException;
 import org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus;
+import org.dcsa.skernel.errors.exceptions.ConcreteRequestErrorMessageException;
 
 import static org.dcsa.edocumentation.domain.persistence.entity.enums.BkgDocumentStatus.*;
 
-public class BookingStateMachineImpl implements BookingStateMachine {
+public class BookingStateMachineImpl extends AbstractStateMachine<BkgDocumentStatus> implements BookingStateMachine {
```
Comment (pimg) [inline edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/booking/BookingStateMachineImpl.java:12]: what's the benefit of having a separate BookingStateMachine interface and impl? Are we expecting more implementations?

Comment (nt-as) [reply edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/booking/BookingStateMachineImpl.java:12]: Primarily that it was easier to see what was going on (i.e., what methods do the state machine have).  I can collapse them into a single class if you do not feel the (current) implementation is distracting.

Comment (pimg) [reply edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/booking/BookingStateMachineImpl.java:12]: I think now that other parts have been separated these can be collapsed into a single class.

Comment (nt-as) [reply edocumentation-domain/src/main/java/org/dcsa/edocumentation/domain/booking/BookingStateMachineImpl.java]: Fixed

**Review by pimg — APPROVED** [review]

**Pull request merged**